### PR TITLE
fixes: multiple fixes

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

Based on recent mistakes I'm forcing  if the description is not found for an arg we now return an error

This also adds initial testing

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cobra2snooty.go [hunk 1]
```diff
@@ -34,13 +34,13 @@ const (
 	defaultExtension = ".txt"
 )
 
-// GenSnootyTree generates the docs for the full tree of commands.
-func GenSnootyTree(cmd *cobra.Command, dir string) error {
```
Comment (gssbzn) [inline cobra2snooty.go:38]: the usage of this is `cobra2snooty.GenSnootyTree` in the spirit of go don't stutter, `cobra2snooty.GenTreeDocs` is a better name

File: cobra2snooty.go [hunk 2]
```diff
@@ -109,8 +109,10 @@ func GenSnootyCustom(cmd *cobra.Command, w io.Writer) error {
 		buf.WriteString(syntaxHeader)
 		buf.WriteString(fmt.Sprintf("\n   %s\n\n", strings.ReplaceAll(cmd.UseLine(), "[flags]", "[options]")))
 	}
-	printArgsSnooty(buf, cmd)
-	printOptionsSnooty(buf, cmd)
+	if err := printArgs(buf, cmd); err != nil {
+		return err
+	}
+	printOptions(buf, cmd)
```
Comment (colm-quinn) [inline cobra2snooty.go:115]: 💯 Love this

File: cobra2snooty.go [hunk 3]
```diff
@@ -176,53 +178,3 @@ type byName []*cobra.Command
...
-   :header-rows: 1
-   :widths: 20 10 10 60
-
-   * - Name
-     - Type
-     - Required
-     - Description
-`
-
-func printArgsSnooty(buf *bytes.Buffer, cmd *cobra.Command) {
```
Comment (gssbzn) [inline cobra2snooty.go:190]: moved to a new file

File: options.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+			requiredSlice = strings.Split(requiredArgs, ",")
+		}
+
+		for _, arg := range strings.Split(args, ",") {
+			required := stringInSlice(requiredSlice, arg)
+			if description, hasDescription := cmd.Annotations[arg+"Desc"]; hasDescription {
+				line := fmt.Sprintf("   * - %s\n     - string\n     - %v\n     - %s", arg, required, description)
+				buf.WriteString(line)
+			} else {
+				return fmt.Errorf("%w: %s", ErrMissingDescription, arg)
```
Comment (gssbzn) [inline options.go:40]: if we can't find a description error the doc generation

Comment (colm-quinn) [reply options.go:40]: 💯

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for this change, I can see it catching a lot of issues.

**Comment by gssbzn** [comment]
> I can see it catching a lot of issues.

I wrote this things I was using it wrong so I can confirm 😛

File: options.go (hunk 4 above)
Comment (gssbzn) [reply options.go:40]: cc @andreaangiolillo

File: cobra2snooty.go [hunk 5]
```diff
@@ -34,13 +34,13 @@ const (
 	defaultExtension = ".txt"
 )
 
-// GenSnootyTree generates the docs for the full tree of commands.
-func GenSnootyTree(cmd *cobra.Command, dir string) error {
+// GenTree generates the docs for the full tree of commands.
+func GenTreeDocs(cmd *cobra.Command, dir string) error {
```
Comment (andreaangiolillo) [inline cobra2snooty.go:38]:
```suggestion
// GenTreeDocs generates the docs for the full tree of commands.
func GenTreeDocs(cmd *cobra.Command, dir string) error {
```

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**